### ISSUE TITLE
Use ISO 8601 standard for date

### ISSUE THREAD

**Issue by ChristopherMayes** [ComputationalRadiationPhysics/openPMD-standard#220]
The base standard has:   
- `date`
    - type: *(string)*
    - description: date of creation in format "YYYY-MM-DD HH:mm:ss tz"
    - example: `2015-12-02 17:48:42 +0100`

But there is a good standard for this:
https://en.wikipedia.org/wiki/ISO_8601

I suggest that ISO 8601 is the standard for the `date` string, e.g.
  `2020-05-13T19:41:34-07:00`

**Comment by ax3l** [comment]
Good catch, I was thinking the same thing last week when someone posted about ISO 8601.

**Comment by ax3l** [comment]
Out of the [extended formats](https://www.w3.org/TR/NOTE-datetime) in ISO 8601, should we go for
```
   Complete date plus hours, minutes and seconds:
      YYYY-MM-DDThh:mm:ssTZD (eg 1997-07-16T19:20:30+01:00)
```
or
```
   Complete date plus hours, minutes, seconds and a decimal fraction of a
second
      YYYY-MM-DDThh:mm:ss.sTZD (eg 1997-07-16T19:20:30.45+01:00)
```

Looks like the decimal fraction of a second can be arbitrarily long, which is good.

**Comment by ChristopherMayes** [comment]
I usually use `datetime` builtin Python package to deal with these strings: 
```python
datetime.fromisoformat('1997-07-16T19:20:30+01:00')
```

But @ax3l your second string is not valid.
